mlir/lib/IR/AttributeDetail.h

```diff
@@ -349,6 +350,46 @@ struct StringAttrStorage : public AttributeStorage {
   Dialect *referencedDialect;
 };
 
+//===----------------------------------------------------------------------===//
+// DistinctAttr
+//===----------------------------------------------------------------------===//
+
+/// An attribute to store a distinct reference to another attribute.
+struct DistinctAttrStorage : public AttributeStorage {
+  using KeyTy = std::tuple<Attribute>;
```

Comment:
Is it enforced that this is a tuple?

Reply:
I will double check. Tablegen always produces a tuple and the replace sub element code seems to rely on it. But maybe it can handle a pure attribute as well.

Reply:
yep works 👍